File: Q16.cpp
class Solution {
public:
    int threeSumClosest(vector<int>& nums, int target) {
        int minDist=INT_MAX;
        int ans=0;
        sort(nums.begin(),nums.end());
        for(int i=0;i<nums.size()-2;i++)
        {
            int j=i+1;
            int k=nums.size()-1;
            while(j<k)
            {
                int sum=nums[i]+nums[j]+nums[k];
                if(sum==target)return target;
                else if(sum>target) k--;
                else j++;
                if(minDist>abs(sum-target)){
                    ans=sum;
                    minDist=abs(sum-target);
                }                
            }
        }
        return ans;
    }
};

File: Q16.py
import sys
class Solution:
    def threeSumClosest(self, nums: List[int], target: int) -> int:
        nums.sort()
        ans=0
        diff=sys.maxsize
        n=len(nums)
        for i in range(0,n-2):
            j=i+1
            k=n-1
            while j<k:
                sum1=nums[i]+nums[j]+nums[k]
                if sum1==target:
                    return target
                elif sum1>target:
                    k-=1
                else:
                    j+=1
                
                if diff>abs(sum1-target):
                    ans=sum1
                    diff=abs(sum1-target)
        return ans
                
        
